packages/material-ui/src/styles/withStyles.d.ts

```diff
@@ -37,9 +37,14 @@ export interface WithStylesOptions<ClassKey extends string = string> extends JSS
 
 export type ClassNameMap<ClassKey extends string = string> = Record<ClassKey, string>;
 
-export interface WithStyles<ClassKey extends string = string> extends Partial<WithTheme> {
-  classes: ClassNameMap<ClassKey>;
-}
+export type WithStyles<T extends string | StyleRules | StyleRulesCallback> = Partial<WithTheme> & {
```

Comment:
This is absolutely awesome and a good use case for conditional types 🤩

Reply:
Removing the default `= string` is a breaking change though?

Reply:
@geirsagberg right you are. I've opened #11808 to fix that.